### PR TITLE
feat(query): enable sse4.2 in x86-64 releaser

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Most modern x86-64 based CPU has supported `sse4.2`, we can enable this feature flag.

Closes #issue

### CODE REVIEW

**Comment by sundy-li** [comment]
@everpcpc  Where to add  target-feature=+sse4.2" in Linux releaser ?

```
RUSTFLAGS="-C target-feature=+sse4.2"  cargo build  ..

```

**Comment by everpcpc** [comment]
we can  add rustflags for gnu just like musl here

https://github.com/datafuselabs/databend/blob/main/.github/workflows/databend-release.yml#L172-L176

File: .github/workflows/databend-release.yml [hunk 1]
```diff
@@ -172,14 +172,16 @@ jobs:
...
+          flags="-C target-feature=+sse4.2,link-arg=-Wl,--compress-debug-sections=zlib-gabi"
           echo "RUSTFLAGS=${flags}" >> $GITHUB_ENV
       - name: Build Binary
         run: |
           cargo build --release --target=${{ steps.target.outputs.target }}
       - name: Copyobj zlib for gnu binaries
         if: matrix.platform == 'gnu'
         run: |
+          flags="-C target-feature=+sse4.2"
+          echo "RUSTFLAGS=${flags}" >> $GITHUB_ENV
```
Comment (everpcpc) [inline .github/workflows/databend-release.yml:184]: this env should goes before cargo build

File: .github/workflows/databend-release.yml [hunk 2]
```diff
@@ -170,10 +170,14 @@ jobs:
         with:
           image: ${{ steps.target.outputs.target }}
       - name: Set musl rustflags
-        if: matrix.platform == 'musl'
         run: |
-          flags="-C link-arg=-Wl,--compress-debug-sections=zlib-gabi"
-          echo "RUSTFLAGS=${flags}" >> $GITHUB_ENV
+          if [[ ${{ matrix.platform }} == “musl” ]]; then
```
Comment (Xuanwo) [inline .github/workflows/databend-release.yml:174]: image

Seems weired.

**Review by everpcpc — APPROVED** [review]

File: .github/workflows/databend-release.yml [hunk 3]
```diff
@@ -174,6 +174,16 @@ jobs:
         run: |
           flags="-C link-arg=-Wl,--compress-debug-sections=zlib-gabi"
           echo "RUSTFLAGS=${flags}" >> $GITHUB_ENV
+      - name: Set musl_x86_64 rustflags
+        if: matrix.platform == 'musl' && matrix.arch = 'x86_64'
```
Comment (Xuanwo) [inline .github/workflows/databend-release.yml:178]:
```suggestion
        if: matrix.platform == 'musl' && matrix.arch == 'x86_64'
```

File: .github/workflows/databend-release.yml [hunk 4]
```diff
@@ -174,6 +174,16 @@ jobs:
...
+        run: |
+          flags="-C target-feature=+sse4.2,link-arg=-Wl,--compress-debug-sections=zlib-gabi"
+          echo "RUSTFLAGS=${flags}" >> $GITHUB_ENV
+      - name: Set gnu_x86_64 rustflags
+        if: matrix.platform == 'gnu' && matrix.arch = 'x86_64'
```
Comment (Xuanwo) [inline .github/workflows/databend-release.yml:183]:
```suggestion
        if: matrix.platform == 'gnu' && matrix.arch == 'x86_64'
```

**Review by Xuanwo — APPROVED** [review]

File: .github/workflows/databend-release.yml [hunk 5]
```diff
@@ -174,6 +174,16 @@ jobs:
...
+      - name: Set musl_x86_64 rustflags
+        if: matrix.platform == 'musl' && matrix.arch == 'x86_64'
+        run: |
+          flags="-C target-feature=+sse4.2,link-arg=-Wl,--compress-debug-sections=zlib-gabi"
+          echo "RUSTFLAGS=${flags}" >> $GITHUB_ENV
+      - name: Set gnu_x86_64 rustflags
+        if: matrix.platform == 'gnu' && matrix.arch == 'x86_64'
+        run: |
+          flags="-C target-feature=+sse4.2"
+          echo "RUSTFLAGS=${flags}" >> $GITHUB_ENV
```
Comment (everpcpc) [inline .github/workflows/databend-release.yml:186]:
```suggestion
          echo "RUSTFLAGS=${flags}" >> $GITHUB_ENV
```

**Pull request merged**